src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLFunctionsIT.java

```diff
@@ -159,6 +164,100 @@ public void caseChangeWithAggregationTest() throws IOException {
+
+        checkSuccessfulFieldCast(query, "cast_balance", "STRING");
+    }
+
+    @Test
+    public void castIntFieldToStringWithAliasTest() throws IOException {
+        String query = "SELECT CAST(balance AS STRING) AS cast_string_alias FROM " + TestsConstants.TEST_INDEX_ACCOUNT +
+                       " LIMIT 5";
+
+        checkSuccessfulFieldCast(query, "cast_string_alias", "STRING");
+    }
+
+    @Test
+    public void castIntFieldToFloatWithoutAliasJdbcFormatTest() {
+        JSONObject response = executeJdbcRequest(
+                "SELECT CAST(balance AS FLOAT) FROM " + TestsConstants.TEST_INDEX_ACCOUNT + " LIMIT 5");
+
+        String float_type_cast = "{\"name\":\"cast_balance\",\"type\":\"float\"}";
+        assertEquals(response.getJSONArray("schema").get(0).toString(), float_type_cast);
+    }
+
+    @Test
+    public void castIntFieldToFloatWithAliasJdbcFormatTest() {
+        JSONObject response = executeJdbcRequest(
+                "SELECT CAST(balance AS FLOAT) AS jdbc_float_alias " +
+                        "FROM " + TestsConstants.TEST_INDEX_ACCOUNT + " limit 5");
+
+        String float_type_cast = "{\"name\":\"jdbc_float_alias\",\"type\":\"float\"}";
+        assertEquals(response.getJSONArray("schema").get(0).toString(), float_type_cast);
+    }
+
+    @Test
+    public void castIntFieldToDoubleWithoutAliasOrderByTest() throws IOException {
+        String query = "SELECT CAST(age AS DOUBLE) FROM " + TestsConstants.TEST_INDEX_ACCOUNT + " ORDER BY age LIMIT 5";
+
+        checkSuccessfulFieldCast(query, "cast_age", "DOUBLE");
+    }
+
+    @Test
+    public void castIntFieldToDoubleWithAliasOrderByTest() throws IOException {
+        String query = "SELECT CAST(age AS DOUBLE) AS alias FROM " + TestsConstants.TEST_INDEX_ACCOUNT +
+                       " ORDER BY alias LIMIT 5";
+
+        checkSuccessfulFieldCast(query, "alias", "DOUBLE");
+    }
+
+    @Test
+    public void castIntFieldToFloatWithoutAliasJdbcFormatGroupByTest() {
+        JSONObject response = executeJdbcRequest(
+                "SELECT CAST(balance AS FLOAT) AS jdbc_float_alias " +
+                        "FROM " + TestsConstants.TEST_INDEX_ACCOUNT + " GROUP BY balance LIMIT 5");
+
+        String float_type_cast = "{\"name\":\"balance\",\"type\":\"long\"}";
+        assertEquals(response.getJSONArray("schema").get(0).toString(), float_type_cast);
+    }
+
+    @Test
+    public void castIntFieldToFloatWithAliasJdbcFormatGroupByTest() {
+        JSONObject response = executeJdbcRequest(
+                "SELECT CAST(balance AS FLOAT) AS jdbc_float_alias " +
+                        "FROM " + TestsConstants.TEST_INDEX_ACCOUNT + " GROUP BY jdbc_float_alias LIMIT 5");
+
+        String float_type_cast = "{\"name\":\"jdbc_float_alias\",\"type\":\"float\"}";
+        assertEquals(response.getJSONArray("schema").get(0).toString(), float_type_cast);
+    }
+
+    @Test
+    public void castIntFieldToDoubleWithAliasJdbcFormatGroupByTest() {
```

Comment:
I don't see any result value checks. Could you add them?

For example, cast some float to int and check that the output has correct value (do we round, remove digits after decimal point)?

What will happen if I will try cast age as datetime? string as int?

Reply:
Yes- will add